akka-actor-typed-tests/src/test/scala/docs/akka/typed/supervision/SupervisionCompileOnly.scala

```diff
@@ -36,6 +36,12 @@ object SupervisionCompileOnly {
   Behaviors
     .supervise(Behaviors.supervise(behavior).onFailure[IllegalStateException](SupervisorStrategy.restart))
     .onFailure[IllegalArgumentException](SupervisorStrategy.stop)
+
+  // or flatten ways
+  Behaviors
+    .supervise(behavior)
+    .whenFailure[IllegalStateException](SupervisorStrategy.restart)
+    .whenFailure[IllegalArgumentException](SupervisorStrategy.stop)
```

Comment:
We should show this as the main way now that we have it, not sure we need to show the other one at all